widget/widget.go

```diff
@@ -13,51 +15,95 @@ type BaseWidget struct {
 	position fyne.Position
 	Hidden   bool
 
-	impl fyne.Widget
+	impl         fyne.Widget
+	propertyLock sync.RWMutex
 }
 
 // ExtendBaseWidget is used by an extending widget to make use of BaseWidget functionality.
 func (w *BaseWidget) ExtendBaseWidget(wid fyne.Widget) {
+	w.propertyLock.Lock()
+	defer w.propertyLock.Unlock()
```

Comment:
Use SetFieldsAndRefresh(f func(){...}) for consistency

Reply:
But we do not need to refresh

Reply:
Define SetFields(func()) which SetFieldsAndRefresh(func(), fyne.Widget) uses